RecoLocalCalo/HcalRecProducers/src/HBHEPhase1Reconstructor.cc

```diff
@@ -0,0 +1,513 @@
+        const bool makeThisRechit = !channelInfo->isDropped();
+        if (infos && (saveDroppedInfos_ || makeThisRechit))
+            infos->push_back(*channelInfo);
+
+        // Reconstruct the rechit
+        if (rechits && makeThisRechit)
+        {
+            const HcalRecoParam* pptr = nullptr;
+            if (recoParamsFromDB_)
+                pptr = param_ts;
+            HBHERecHit rh = reco_->reconstruct(*channelInfo, pptr, calib, isRealData);
+            if (rh.id().rawId())
+            {
+                setAsicSpecificBits(frame, *channelInfo, &rh);
+                setCommonStatusBits(*channelInfo, &rh);
+                rechits->push_back(rh);
+            }
+        }
+    }
+}
+
+void HBHEPhase1Reconstructor::setCommonStatusBits(
+    const HBHEChannelInfo& info, HBHERecHit* rh)
+{
+    // FIX THIS!!! after status bit inventory
+}
+
+void HBHEPhase1Reconstructor::setAsicSpecificBits(
+    const HBHEDataFrame& frame, const HBHEChannelInfo& info, HBHERecHit* rh)
+{
+    // FIX THIS!!! after status bit inventory
+}
+
+void HBHEPhase1Reconstructor::setAsicSpecificBits(
+    const QIE11DataFrame& frame, const HBHEChannelInfo& info, HBHERecHit* rh)
+{
+    // FIX THIS!!! after status bit inventory
+}
+
+// ------------ method called to produce the data  ------------
+void
+HBHEPhase1Reconstructor::produce(edm::Event& e, const edm::EventSetup& eventSetup)
+{
+    using namespace edm;
+
+    // Get the Hcal topology if needed
+    ESHandle<HcalTopology> htopo;
+    if (tsFromDB_ || recoParamsFromDB_)
+    {
+        eventSetup.get<HcalRecNumberingRecord>().get(htopo);
+        paramTS_->setTopo(htopo.product());
+    }
+
+    // Fetch the calibrations
+    ESHandle<HcalDbService> conditions;
+    eventSetup.get<HcalDbRecord>().get(conditions);
+
+    ESHandle<HcalChannelQuality> p;
+    eventSetup.get<HcalChannelQualityRcd>().get("withTopo", p);
+ 
+    ESHandle<HcalSeverityLevelComputer> mycomputer;
+    eventSetup.get<HcalSeverityLevelComputerRcd>().get(mycomputer);
+
+    // Find the input data
+    unsigned maxOutputSize = 0;
+    Handle<HBHEDigiCollection> hbDigis;
+    if (processHB_)
+    {
+        e.getByToken(tok_hb_, hbDigis);
+        maxOutputSize += hbDigis->size();
+    }
+
+    Handle<QIE11DigiCollection> heDigis;
+    if (processHE_)
+    {
+        e.getByToken(tok_he_, heDigis);
+        maxOutputSize += heDigis->size();
+    }
+
+    // Create new output collections
+    std::unique_ptr<HBHEChannelInfoCollection> infos;
+    if (saveInfos_)
+    {
+        infos = std::make_unique<HBHEChannelInfoCollection>();
+        infos->reserve(maxOutputSize);
+    }
+
+    std::unique_ptr<HBHERecHitCollection> out;
+    if (makeRecHits_)
+    {
+        out = std::make_unique<HBHERecHitCollection>();
+        out->reserve(maxOutputSize);
+    }
+
+    // Process the input collections, filling the output ones
+    const bool isData = e.isRealData();
+    if (processHB_)
```

Comment:
this will need to be changed in run3 again (and also, likely, to support current setup from one module)

Better define full flexibility now.
``` if processHBHE_QIE8 (up to 2016) -> else if processHB_QIE8 or processHE_QIE11 (2017) -> else (run3) ```